c10/core/SymInt.h

```diff
@@ -364,4 +364,53 @@ DECLARE_SYMINT_OP(size_t, SymInt)
 
 C10_API std::ostream& operator<<(std::ostream& os, const SymInt& s);
 C10_API SymInt operator-(const SymInt& s);
+
+inline bool sym_eq(int64_t a, int64_t b) {
```

Comment:
Any particular reason this set of changes is in this PR ?

Reply:
I need to mark some C++ conditions as oblivious, but to do that, the equality test has to be done symbolically. But the areas I updated are templated logic that either take int or SymInt, so I need some more juice for it